### PR TITLE
Improvements and structure changes for autoscaling docs

### LINKED ISSUE #2375: Productionize and public target burst capacity document.
The document: https://docs.google.com/document/d/1ubszZxikwe0O1JsA-Rjwdv3YA-rgHWKLI9WPAD083ig/edit

I took it to the WG meeting some time ago.
This should give it proper tracking.

/cc @abrennan89

### PR DESCRIPTION
**Pull request by abrennan89**
## Proposed Changes

- Split autoscaling into its own sub directory and smaller topics, to make the content easier to consume.
- General content editing.

### CODE REVIEW

File: docs/serving/autoscaling/autoscaling.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+## Autoscaling concepts
+
+The following sections describe important concepts and terms used when discussing autoscaling in Knative.
+
+* [Global versus per revision settings](./global-vs-per-revision.md)
+* [Supported Autoscaler types](./supported-autoscaler-types.md) (includes configuration settings for enabling HPA)
+* [Metrics and targets](./autoscaling-metrics-targets.md)
+* [Concurrency](./concurrency.md)
+*
```
Comment (mattmoor) [inline docs/serving/autoscaling/autoscaling.md:27]: Remove trailing whitespace:
```suggestion
*
```

Comment (abrennan89) [reply docs/serving/autoscaling/autoscaling.md:27]: I wouldn't bother reviewing this until I'm done with a draft, just FYI :smile:

**Comment by mattmoor** [comment]
/approve
/hold

If you `git commit --amend` and force push, it will stage the docs for you now :D

File: docs/serving/autoscaling/README.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+Knative Serving provides users with the functionality to automatically scale the number of application replicas based on demand.
+
+The component responsible for autoscaling functionality in Knative Serving is the **Autoscaler** component. The Autoscaler watches traffic flow to the application, and scales the application up or down based on metrics.
+
+The default Autoscaler used in Knative deployments is the Knative Pod Autoscaler (KPA). However, Knative deployments can be configured to use the Kubernetes Horizontal Pod Autoscaler (HPA) instead.
+<!-- Question: removed mention of BYO autoscaler since we don't seem to doc this yet - does it need to be added, can we provide instructions for it yet?-->
```
Comment (abrennan89) [inline docs/serving/autoscaling/README.md:6]: @markusthoemmes just wondering if you know the status of BYO autoscaler and whether we want to doc using it yet?

File: docs/serving/autoscaling/supported-autoscaler-types.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+Knative Serving supports the implementation of Knative Pod Autoscaler (KPA) and Kubernetes' Horizontal Pod Autoscaler (HPA). The features and limitations of each of these Autoscalers are listed below.
+
+## KPA
+* Part of the Knative Serving core and enabled by default.
+* Supports scale to zero functionality.
+* Does not support CPU-based autoscaling.
+
+## HPA
+* Not part of the Knative Serving core, and must be enabled during Knative Serving installation.
+<!-- Question: do we provide details in the install docs about this?-->
```
Comment (abrennan89) [inline docs/serving/autoscaling/supported-autoscaler-types.md:17]: @markusthoemmes do we need to add somethng about HPA to the install docs?

File: docs/serving/autoscaling/target-burst-capacity.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+type: "docs"
+---
+
+_Target burst capacity_ is a [global and per-revision](./global-vs-per-revision.md) integer setting.
+
+This value specifies how large of a _burst_ of traffic a Knative application can handle without buffering.
+
+If a burst of traffic is too large for the application to handle, the Knative _Activator_ will be placed in the request path to protect the application revision and optimize request load balancing.
+<!-- Question: is revision accurate here or is it protecting replicas of an app?-->
+<!-- TODO: docs about Activator?-->
```
Comment (abrennan89) [inline docs/serving/autoscaling/target-burst-capacity.md:14]: @vagababov

File: docs/serving/autoscaling/concurrency.md [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+For example, if `containerConcurrency` is set to 10, and the target utilization value is set to 70 (percent), the Autoscaler will create a new replica when the average number of concurrent requests across all existing replicas reaches 7.
+Requests numbered 7 to 10 will still be sent to the existing replicas, but this allows for additional replicas to be started in anticipation of being needed when the `containerConcurrency` limit is reached.
+
+* **Global key:** `container-concurrency-target-percentage`
+* **Per-revision annotation key:** `autoscaling.knative.dev/targetUtilizationPercentage`
+* **Possible values:** float
+* **Default:** `70`
+
+**Note:** If the activator is in the routing path, it will fully load all replicas up to `containerConcurrency`. It currently does not take target utilization into account.
+<!-- TODO: Add docs about TBC, link here?-->
```
Comment (abrennan89) [inline docs/serving/autoscaling/concurrency.md:127]: Need to add docs about target utilization first @abrennan89 TODO

File: docs/serving/autoscaling/README.md (hunk 2 above)
Comment (markusthoemmes) [reply docs/serving/autoscaling/README.md:6]: I don't think so. It's non-trivial and requires it's own set of development documentation.

File: docs/serving/autoscaling/target-burst-capacity.md [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+linkTitle: "Target burst capacity"
+weight: 50
+type: "docs"
+---
+
+_Target burst capacity_ is a [global and per-revision](./global-vs-per-revision.md) integer setting.
+
+This value specifies how large of a _burst_ of traffic a Knative application can handle without buffering.
+
+If a burst of traffic is too large for the application to handle, the Knative _Activator_ will be placed in the request path to protect the application revision and optimize request load balancing.
```
Comment (vagababov) [inline docs/serving/autoscaling/target-burst-capacity.md:12]:
```suggestion
If a desired burst of traffic is too large for the application to handle at current scale, the Knative _Activator_ will be placed in the request path to protect the application revision and optimize request load balancing.
```

File: docs/serving/autoscaling/target-burst-capacity.md [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+<!-- Question: is revision accurate here or is it protecting replicas of an app?-->
```
Comment (vagababov) [inline docs/serving/autoscaling/target-burst-capacity.md:13]: I think revision==app, kind of.

File: docs/serving/autoscaling/target-burst-capacity.md (hunk 4 above)
Comment (vagababov) [reply docs/serving/autoscaling/target-burst-capacity.md:14]: Activator for most intents and purposes is detail of implementation here. So I am not sure we want to add more docs here.

File: docs/serving/autoscaling/supported-autoscaler-types.md (hunk 3 above)
Comment (markusthoemmes) [reply docs/serving/autoscaling/supported-autoscaler-types.md:17]: AFAIK, there's a section explaining how to install it there already.

Comment (markusthoemmes) [reply docs/serving/autoscaling/supported-autoscaler-types.md:17]: You could link here https://knative.dev/docs/install/any-kubernetes-cluster/#optional-serving-extensions

File: docs/serving/autoscaling/README.md [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+Knative Serving revisions are configured by default to use generic settings that are suitable for the majority of use cases. However, some workloads may required a custom, more finely-tuned configuration.
+This guide provides information about configuration options that you can modify to fit the requirements of your workload.
+
+## Autoscaling concepts
+
+The following sections describe important concepts and terms used when discussing autoscaling in Knative, as well as configuration details where applicable.
+
+* [Supported Autoscaler types](./supported-autoscaler-types.md) (includes configuration settings for enabling HPA)
+* [Global versus per revision settings](./global-vs-per-revision.md)
+* [KPA specific behavior](./kpa-specific.md)
```
Comment (markusthoemmes) [inline docs/serving/autoscaling/README.md:16]: I would move this to the bottom of the list. I'd expect users to care about all the other bits first.

File: docs/serving/autoscaling/concurrency.md [hunk 9]
```diff
@@ -0,0 +1,223 @@
...
+  name: knative-serving
+spec:
+  config:
+    autoscaler:
+      container-concurrency-target-percentage: "80"
+```
+{{< /tab >}}
+{{< /tabs >}}
+
+## Requests-per-second Target
```
Comment (markusthoemmes) [inline docs/serving/autoscaling/concurrency.md:172]: This is not related to concurrency.

File: docs/serving/autoscaling/autoscaling-metrics.md [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+
+The `target` annotation determines the value that the Autoscaler tries to maintain for the specified `metric` of every replica of an application. `target` annotations per-revision are "metric agnostic", meaning that the target is simply an integer value, which can be applied for any `metric` type.
+
+For example, if the `metric` annotation of a revision is set to [`concurrency`](./concurrency.md), and the `target` annotation has a value of 10, the Autoscaler attempts to ensure that every replica receives 10 requests at a time.
+
+For [global](./global-vs-per-revision.md) configurations using the default [KPA](./supported-autoscaler-types.md) Autoscaler, there are two keys that can be defined to determine the target value for a specific supported metric type.
+These keys are `container-concurrency-target-default` and `requests-per-second-target-default`, since global KPA configurations only support the `concurrency` and `requests-per-second` (rps) metrics.
+
+Deployments using the [HPA](./supported-autoscaler-types.md) Autoscaler can only configure metrics per revision, and only support the `cpu` metric.
```
Comment (markusthoemmes) [inline docs/serving/autoscaling/autoscaling-metrics.md:17]: This is not true. The HPA supports concurrency and rps as well. It supports CPU in addition to that. The KPA only supports the former two.

File: docs/serving/autoscaling/README.md [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+* [Metrics and targets](./autoscaling-metrics-targets.md)
+* [Concurrency](./concurrency.md)
+* [Scale bounds](./scale-bounds.md)
+* [Target burst capacity](./target-burst-capacity.md)
```
Comment (markusthoemmes) [inline docs/serving/autoscaling/README.md:20]: I wonder if this should be under "Autoscaling concepts". I intended to doc it under a bigger arch, explaining "Loadbalancing concepts". @vagababov already started a doc in that direction IIRC.

File: docs/serving/autoscaling/autoscaling-metrics.md [hunk 12]
```diff
@@ -0,0 +1,46 @@
+---
+title: "Metrics and targets"
```
Comment (markusthoemmes) [inline docs/serving/autoscaling/autoscaling-metrics.md:2]: Do we really want to mix metrics and targets here and then pick targets up again in the concurrency (and RPS) chapter?

File: docs/serving/autoscaling/target-burst-capacity.md [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+      containerConcurrency: 4
+      containers:
+      - env:
+        - name: TARGET
+          value: T2
+        image: "your docker image"
+        imagePullPolicy: Always
+        name: user-container
+      timeoutSeconds: 300
+```
```
Comment (markusthoemmes) [inline docs/serving/autoscaling/target-burst-capacity.md:57]: Let's reuse the same example as for all other examples. This contains a lot of fields that are not required at all. Only the `targetBurstCapacity` setting is significant for these docs.

File: docs/serving/autoscaling/target-burst-capacity.md (hunk 6 above)
Comment (abrennan89) [reply docs/serving/autoscaling/target-burst-capacity.md]: It's actually in the Google developer docs style guide that we shouldn't use "desire", and I think this would sort of just add fluff anyway.

**Comment by abrennan89** [comment]
> I'm no expert but I'm wondering if making the users click more to get to the respective chapters is doing more harm than it does good. I personally like docs like these, which are mostly setting centered, to be on a single page so I can CTRL-F search for what I need.

The idea is that each section should have a topic that's a self-explanatory user story, i.e. a user looking for how to configure a certain thing will go to the section for that thing, rather than needing to search through a lengthy page/doc to find it.
It also IMHO makes documentation a lot easier to maintain.

**Comment by abrennan89** [comment]
@markusthoemmes my to do list for Monday:
- Update what's supported by KPA vs HPA
- Add a new issue to track the load balancing work and move any sections after that need to go there (@vagababov is still drafting the doc I think)
- Split out metrics and targets and explain them better.
- Clean up examples.

Lmk if there's anything else. Thanks!

File: docs/serving/autoscaling/README.md (hunk 11 above)
Comment (abrennan89) [reply docs/serving/autoscaling/README.md]: Added a follow up issue for loadbalancing https://github.com/knative/docs/issues/2477

File: docs/serving/autoscaling/autoscaling-concepts.md [hunk 14]
```diff
@@ -0,0 +1,135 @@
...
+  name: knative-serving
+spec:
+  config:
+    autoscaler:
+      pod-autoscaler-class: "kpa.autoscaling.knative.dev"
+```
+{{< /tab >}}
+{{< /tabs >}}
+
+## Global versus per-revision settings
```
Comment (vagababov) [inline docs/serving/autoscaling/autoscaling-concepts.md:81]: This should probably go above the class, since it affects the choices made there as well.

File: docs/serving/autoscaling/autoscaling-concepts.md [hunk 15]
```diff
@@ -0,0 +1,135 @@
...
+ container-concurrency-target-default: "100"
+ container-concurrency-target-percentage: "0.7"
+ enable-scale-to-zero: "true"
+ max-scale-up-rate: "1000"
+ max-scale-down-rate: "2"
+ panic-window-percentage: "10"
+ panic-threshold-percentage: "200"
+ scale-to-zero-grace-period: "30s"
+ stable-window: "60s"
+ tick-interval: "2s"
```
Comment (vagababov) [inline docs/serving/autoscaling/autoscaling-concepts.md:110]: `tick-interval` is removed.

File: docs/serving/autoscaling/autoscaling-concepts.md [hunk 16]
```diff
@@ -0,0 +1,135 @@
...
+ namespace: knative-serving
+data:
+ container-concurrency-target-default: "100"
+ container-concurrency-target-percentage: "0.7"
+ enable-scale-to-zero: "true"
+ max-scale-up-rate: "1000"
+ max-scale-down-rate: "2"
+ panic-window-percentage: "10"
+ panic-threshold-percentage: "200"
+ scale-to-zero-grace-period: "30s"
```
Comment (vagababov) [inline docs/serving/autoscaling/autoscaling-concepts.md:108]: Also new one:
`scale-to-zero-pod-retention-period: "0s"`.

File: docs/serving/autoscaling/target-burst-capacity.md [hunk 17]
```diff
@@ -0,0 +1,47 @@
...
+* Setting the target burst capacity per revision.
+
+## Setting the target burst capacity per revision
+
+* **Global key:** No global key.
+* **Per-revision annotation key:** `autoscaling.knative.dev/targetBurstCapacity`
+* **Possible values:** float
+* **Default:** `70`
+
+**Note:** If the activator is in the routing path, it will fully load all replicas up to `containerConcurrency`. It currently does not take target utilization into account.
```
Comment (vagababov) [inline docs/serving/autoscaling/target-burst-capacity.md:27]:
```suggestion
**Note:** If the activator is in the routing path, it will fully load all replicas up to `containerConcurrency`. It currently applies target utilization only on revision level.
```
Or somewhere thereof. Since the integral utilization is respected, per pod one isn't

File: docs/serving/autoscaling/README.md [hunk 18]
```diff
@@ -0,0 +1,18 @@
...
+
+For more information about which metrics can be used to control the Autoscaler, see the [metrics](./autoscaling-metrics.md) documentation.
+
+## Optional autoscaling configuration tasks
+
+* Configure your Knative deployment to use the Kubernetes [Horizontal Pod Autoscaler (HPA)](../../install/any-kubernetes-cluster.md#optional-serving-extensions) instead of the default KPA.
+* Disable scale to zero functionality for your cluster ([global configuration only](./scale-to-zero.md)).
+* Configure the [type of metrics](./autoscaling-metrics.md) your Autoscaler consumes.
+* Configure [concurrency limits](./concurrency.md) for applications.
+* Try out the [Go Autoscale Sample App](./autoscale-go/README.md).
```
Comment (evankanderson) [inline docs/serving/autoscaling/README.md:18]: Do you see doing a similar move for other samples in the future (e.g. traffic splitting samples into the networking subsection)?

File: docs/serving/autoscaling/autoscale-go/index.md [hunk 19]
```diff
@@ -1,7 +1,7 @@
 ---
 title: "Autoscale Sample App - Go"
-linkTitle: "Autoscaling - Go"
-weight: 1
+linkTitle: "Autoscale Sample App - Go"
+weight: 100
 type: "docs"
```
Comment (evankanderson) [inline docs/serving/autoscaling/autoscale-go/index.md:5]: You may want to add an `aliases` here to allow the old links to also work.

https://gohugo.io/content-management/urls/#aliases

File: docs/serving/autoscaling/autoscaling-concepts.md (hunk 14 above)
Comment (evankanderson) [reply docs/serving/autoscaling/autoscaling-concepts.md:81]: I think this fits here as well -- it provides more detail, but if users are happy with the first choice they found, it should work.

File: docs/serving/autoscaling/autoscaling-concepts.md [hunk 20]
```diff
@@ -0,0 +1,135 @@
...
+Configuring for autoscaling in Knative can be set using either global or per-revision settings.
+
+1. If no per-revision autoscaling settings are specified, the global settings will be used.
+1. If per-revision settings are specified, these will override the global settings when both types of settings exist.
+
+### Global settings
+
+Global settings for autoscaling are configured using the `config-autoscaler` ConfigMap. If you installed Knative Serving using the Operator, you can set global configuration settings in the `spec.config.autoscaler` ConfigMap, located in the `KnativeServing` custom resource (CR).
+
+#### Example of the default Kubernetes ConfigMap
```
Comment (evankanderson) [inline docs/serving/autoscaling/autoscaling-concepts.md:92]:
```suggestion
#### Example of the default autoscaling ConfigMap
```

"Kubernetes ConfigMap" makes me think that this is doing something to Kubernetes, rather than simply stored in Kubernetes.

File: docs/serving/autoscaling/autoscaling-concepts.md [hunk 21]
```diff
@@ -0,0 +1,135 @@
...
+ scale-to-zero-grace-period: "30s"
+ stable-window: "60s"
+ tick-interval: "2s"
+ target-burst-capacity: "200"
+ requests-per-second-target-default: "200"
+```
+
+### Per-revision settings
+
+Per-revision settings for autoscaling are configured by adding _annotations_ to a revision.
```
Comment (evankanderson) [inline docs/serving/autoscaling/autoscaling-concepts.md:117]: Are the annotation keys the same ones as are present in the ConfigMap?

File: docs/serving/autoscaling/autoscaling-metrics.md [hunk 22]
```diff
@@ -0,0 +1,72 @@
...
+
+The metric configuration defines which metric type is watched by the Autoscaler.
+
+## Setting metrics per revision
+
+For [per-revision](./autoscaling-concepts.md) configuration, this is determined using the `autoscaling.knative.dev/metric` annotation.
+The possible metric types that can be configured per revision depend on the type of Autoscaler implementation you are using:
+
+* The default KPA Autoscaler supports the `concurrency` and `rps` metrics.
+* The HPA Autoscaler supports the `concurrency`, `rps` and `cpu` metrics.
```
Comment (evankanderson) [inline docs/serving/autoscaling/autoscaling-metrics.md:16]: Can you describe what these options mean?

Comment (evankanderson) [reply docs/serving/autoscaling/autoscaling-metrics.md:16]: In particular, how concurrency and RPS differ.

Feel free to defer with a `<!-- TODO: ... -->` if you don't have bandwidth right now.

File: docs/serving/autoscaling/autoscaling-targets.md [hunk 23]
```diff
@@ -0,0 +1,36 @@
...
+See the [metrics](./autoscaling-metrics.md) documentation for more information about configurable metric types.
+
+The `target` annotation, used to configure per-revision targets,  is _metric agnostic_. This means the target is simply an integer value, which can be applied for any metric type.
+
+## Configuring targets
+
+* **Global settings key:** `container-concurrency-target-default` for setting a concurrency target, and `requests-per-second-target-default` for setting a requests-per-second (RPS) target. For more information, see the documentation on [metrics](./autoscaling-metrics.md).
+* **Per-revision annotation key:** `autoscaling.knative.dev/target`
+* **Possible values:** An integer (metric agnostic).
+* **Default:** `"50"` for the `target` annotation, `"100"` for `container-concurrency-target-default`, and `"200"` for `requests-per-second-target-default`.
```
Comment (evankanderson) [inline docs/serving/autoscaling/autoscaling-targets.md:18]: I'm not sure I understand how the `target` and `-default` defaults work together?

@vagababov

File: docs/serving/autoscaling/autoscaling-targets.md [hunk 24]
```diff
@@ -0,0 +1,36 @@
...
+<!-- Is this the correct default value for target?-->
+
+{{< tabs name="Configuring targets" default="Per-revision example" >}}
+{{% tab name="Per-revision example using target annotation" %}}
```
Comment (evankanderson) [inline docs/serving/autoscaling/autoscaling-targets.md:22]: Only one tab intended?

File: docs/serving/autoscaling/concurrency.md [hunk 25]
```diff
@@ -0,0 +1,160 @@
+---
+title: "Configuring concurrency"
+linkTitle: "Configuring concurrency"
+weight: 40
+type: "docs"
+---
+
+Concurrency determines the number of simultaneous requests that can be processed by each replica of an application at any given time.
+<!-- this is where including files would be useful. We could create a concurrency global config module and insert it here, in the docs for metrics, and in the docs for targets. Showing the correct information each time instead of having it in one place with the per revision config jumbled in with it makes it easier to understand IMHO, and would mean users don't need to visit different pages or hunt for the same information for similar user stories @abrennan89.-->
```
Comment (evankanderson) [inline docs/serving/autoscaling/concurrency.md:9]: Reference an issue in the website repo if this isn't working and you want it to?

File: docs/serving/autoscaling/concurrency.md [hunk 26]
```diff
@@ -0,0 +1,160 @@
...
+It is possible to set either a _soft_ or _hard_ concurrency limit.
+
+**NOTE:** If both a soft and a hard limit are specified, the smaller of the two values will be used. This prevents the Autoscaler from having a target value that is not permitted by the hard limit value.
+
+The soft limit is a targeted limit rather than a strictly enforced bound. In some situations, particularly if there is a sudden _burst_ of requests, this value can be exceeded.
+
+The hard limit is an enforced upper bound.
+If concurrency reaches the hard limit, surplus requests will be buffered and must wait until enough capacity is free to execute the requests.
+
+**IMPORTANT:** Using a hard limit configuration is only recommended if there is a clear use case for it with your application. Having a low hard limit specified may have a negative impact on the throughput and latency of an application.
```
Comment (evankanderson) [inline docs/serving/autoscaling/concurrency.md:25]: Mention it may cause additional cold starts?

File: docs/serving/autoscaling/autoscaling-targets.md (hunk 23 above)
Comment (markusthoemmes) [reply docs/serving/autoscaling/autoscaling-targets.md:18]: FWIW: There's no default for the target annotation. There are global default settings per metric as shown in the global settings key.

File: docs/serving/autoscaling/autoscaling-concepts.md (hunk 14 above)
Comment (abrennan89) [reply docs/serving/autoscaling/autoscaling-concepts.md:81]: I think this is why we need to be able to reuse / include content - so we can have a blurb about something in more than one place in the docs but maintain it from one file.

File: docs/serving/autoscaling/README.md (hunk 18 above)
Comment (abrennan89) [reply docs/serving/autoscaling/README.md:18]: Yep, I'd like to do this with all of them, it just makes more sense to me as a place where someone would look for it. WDYT?

File: docs/serving/autoscaling/autoscale-go/index.md (hunk 19 above)
Comment (abrennan89) [reply docs/serving/autoscaling/autoscale-go/index.md:5]: Hopefully I've done these right now :slightly_smiling_face:

File: docs/serving/autoscaling/autoscaling-concepts.md (hunk 21 above)
Comment (abrennan89) [reply docs/serving/autoscaling/autoscaling-concepts.md:117]: Nope, they're different afaik, it's explained in the section for each key.

File: docs/serving/autoscaling/autoscaling-metrics.md (hunk 22 above)
Comment (abrennan89) [reply docs/serving/autoscaling/autoscaling-metrics.md:16]: Added a TODO note :) I think I had pretty much the same note previously and removed it because I knew I wouldn't get to it :joy:

File: docs/serving/autoscaling/autoscaling-targets.md (hunk 24 above)
Comment (abrennan89) [reply docs/serving/autoscaling/autoscaling-targets.md:22]: Fixed

File: docs/serving/autoscaling/concurrency.md (hunk 25 above)
Comment (abrennan89) [reply docs/serving/autoscaling/concurrency.md:9]: I'll come back to it when I have more time :)

File: docs/serving/autoscaling/autoscaling-metrics.md (hunk 22 above)
Comment (evankanderson) [reply docs/serving/autoscaling/autoscaling-metrics.md:16]: A TODO lets someone else pick up the item if you don't get back to it. 😁

File: docs/serving/autoscaling/README.md (hunk 18 above)
Comment (evankanderson) [reply docs/serving/autoscaling/README.md:18]: I don't think we've discussed it; I don't have an intrinsic problem with it, but it would be good to have a documented strategy.

File: docs/serving/autoscaling/_index.md [hunk 27]
```diff
@@ -0,0 +1,10 @@
+---
+title: "Autoscaling"
+linkTitle: "Autoscaling"
+weight: 20
+type: "docs"
+aliases:
+    - ./docs/serving/configuring-autoscaling.md
```
Comment (evankanderson) [inline docs/serving/autoscaling/_index.md:7]: I don't think you want this to be `../configuring-autoscaling` or `/docs/serving/configuring-autoscaling`; this would make an alias of `/docs/serving/autoscaling/docs/serving/configuring-autoscaling.md`.

File: docs/serving/autoscaling/autoscale-go/index.md [hunk 28]
```diff
@@ -0,0 +1,10 @@
+---
+title: "Autoscale Sample App - Go"
+linkTitle: "Autoscale Sample App - Go"
+weight: 100
+type: "docs"
+aliases:
+    - ./docs/serving/samples/autoscale-go/README.md
```
Comment (evankanderson) [inline docs/serving/autoscaling/autoscale-go/index.md:7]: Again, I think you want the URL on the site:

`/docs/serving/samples/autoscale-go` or `../../samples/autoscale-go`

File: docs/serving/autoscaling/_index.md [hunk 29]
```diff
@@ -0,0 +1,10 @@
+---
+title: "Autoscaling"
+linkTitle: "Autoscaling"
+weight: 20
+type: "docs"
+aliases:
+    - /docs/serving/autoscaling/docs/serving/configuring-autoscaling.md
```
Comment (evankanderson) [inline docs/serving/autoscaling/_index.md:7]: I think you want this to be:
```suggestion
    - /docs/serving/configuring-autoscaling/
```

So that existing links to https://knative.dev/development/serving/configuring-autoscaling/ will redirect to this page.

Comment (abrennan89) [reply docs/serving/autoscaling/_index.md:7]: Thank you, apologies still getting my head around the aliases :slightly_smiling_face:

**Comment by abrennan89** [comment]
No problem, thanks for looking at this @evankanderson - should be finally ready now :joy:

**Comment by evankanderson** [comment]
BTW, you don't need to squash your changes to the docs repo; Tide will do a squash merge automatically when it merges, so each PR becomes one commit. Leaving the commits un-squashed during review makes it easier to see what's changed since the last commit. (But I'll run through this one and approve in a moment.)
